src/XPurchase/VirtoCommerce.XPurchase/CartAggregateRepository.cs

```diff
@@ -103,7 +103,7 @@ public async Task<SearchCartResponse> SearchCartAsync(ShoppingCartSearchCriteria
             return new SearchCartResponse() { Results = cartAggregates, TotalCount = searchResult.TotalCount };
         }
 
-        public virtual async Task RemoveCartAsync(string cartId) => await _shoppingCartService.DeleteAsync(new[] { cartId });
+        public virtual async Task RemoveCartAsync(string cartId) => await _shoppingCartService.DeleteAsync(new[] { cartId }, softDelete: true);
```

Comment:
you forgot to run Background task for hard deletion

Reply:
Tech debt created for the next sprint.

Reply:
https://virtocommerce.atlassian.net/browse/PT-192